build/source/driver/multi_driver.f90

```diff
@@ -247,77 +357,123 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
+                 err,message)               ! error control
+ call handle_err(err,message)
+
  ! re-calculate height of each layer
  call calcHeight(&
                  ! input/output: data structures
-                 indx_data,   & ! intent(in): layer type
-                 mvar_data,   & ! intent(inout): model variables for a local HRU
+                 indxStruct%hru(iHRU),   & ! intent(in): layer type
+                 progStruct%hru(iHRU),   & ! intent(inout): model prognostic (state) variables for a local HRU
                  ! output: error control
                  err,message); call handle_err(err,message)
- ! compute derived model variables that are pretty much constant over each HRU
- call E2T_lookup(err,message); call handle_err(err,message) ! calculate a look-up table for the temperature-enthalpy conversion
- call rootDensty(err,message); call handle_err(err,message) ! calculate vertical distribution of root density
- call satHydCond(err,message); call handle_err(err,message) ! calculate saturated hydraulic conductivity in each soil layer
- call v_shortcut(err,message); call handle_err(err,message) ! calculate "short-cut" variables such as volumetric heat capacity
+
+ ! calculate vertical distribution of root density
+ call rootDensty(mparStruct%hru(iHRU)%var,& ! vector of model parameters
+                 indxStruct%hru(iHRU),    & ! data structure of model indices
+                 progStruct%hru(iHRU),    & ! data structure of model prognostic (state) variables
+                 diagStruct%hru(iHRU),    & ! data structure of model diagnostic variables
+                 err,message)               ! error control
+ call handle_err(err,message) 
+
+ ! calculate saturated hydraulic conductivity in each soil layer
+ call satHydCond(mparStruct%hru(iHRU)%var,& ! vector of model parameters
+                 indxStruct%hru(iHRU),    & ! data structure of model indices
+                 progStruct%hru(iHRU),    & ! data structure of model prognostic (state) variables
+                 fluxStruct%hru(iHRU),    & ! data structure of model fluxes 
+                 err,message)               ! error control
+ call handle_err(err,message)
+
+ ! calculate "short-cut" variables such as volumetric heat capacity
+ call v_shortcut(mparStruct%hru(iHRU)%var,& ! vector of model parameters
+                 diagStruct%hru(iHRU),    & ! data structure of model diagnostic variables
+                 err,message)               ! error control
+ call handle_err(err,message)
+
  ! overwrite the vegetation height
- HVT(type_data%var(iLookTYPE%vegTypeIndex)) = mpar_data%var(iLookPARAM%heightCanopyTop)
- HVB(type_data%var(iLookTYPE%vegTypeIndex)) = mpar_data%var(iLookPARAM%heightCanopyBottom)
+ HVT(typeStruct%hru(iHRU)%var(iLookTYPE%vegTypeIndex)) = mparStruct%hru(iHRU)%var(iLookPARAM%heightCanopyTop)
+ HVB(typeStruct%hru(iHRU)%var(iLookTYPE%vegTypeIndex)) = mparStruct%hru(iHRU)%var(iLookPARAM%heightCanopyBottom)
+
  ! overwrite the tables for LAI and SAI
```

Comment:
This also happens in line 648 and following. Is there a reason it is done in both locations?